### ISSUE TITLE
Add checks for revert reason strings

### ISSUE THREAD

**Issue by filiplazovic** [ERC725Alliance/erc725#16]
Everywhere we are using `reverting` function replace with our function which will check if revert message is correct. This will help us be confident that our test are correct and our contracts are working as expected.

Depends on #17

**Comment by CPerezz** [comment]
@filiplazovic If I'm not wrong, the Zeppelin Helpers already include this possibility. You can switch between Out of gas reverts, Invalid Opcode reverts and a Simple revert (Which can be caused by a Require statment for example.).

If you refer to for example catch this String "Sender not authorized":
```java
require(
    msg.sender == _account,
    "Sender not authorized."
);
```
On the following code, catch the "Sender not authorized" with an event firing and catching? I think we have tools to do this on the Event helpers of Zeppelin lib.

I've not much time this week to review it. But if this persists, I can take it next week if no one does.

**Comment by filiplazovic** [comment]
So it will be similar to `Simple revert` you mentioned, but with a check for revert message. I didn't planed this to be done via events.
I forgot to mention this feature is available with truffle 5.0.0. See https://github.com/trufflesuite/truffle/releases?after=truffle-blockchain-utils%400.0.6-beta.0
I can create separate issue for updating to truffle 5

**Comment by CPerezz** [comment]
Oh! I wasn't aware of that! Definitely upgrading to truffle 5 will be a nice TODO (if it's stable ofc).

**Comment by filiplazovic** [comment]
Truffle 5 is in beta, but gives us a lot of useful features that will help us write better code. If you are not confident, we can wait for full release.

**Comment by CPerezz** [comment]
The thing is that I think the test should use Web3 1.0 and not Truffle's Web3 version. You know if Truffle includes Web3 1.0 to to the tests? I was gonna make an issue to refactor the tests on web3 1.0 version. But only works with Node 8.X or less. What do you think?

**Comment by filiplazovic** [comment]
Currently, Truffle's web3 version is `0.20.6`. We have web 1.0 installed in our project as a separate package that is used for some things. If you want to change Truffle's default web3 version you would have to install Truffle 5

**Comment by CPerezz** [comment]
It's not necessary, with previous truffle versions it's possible to change the injected Web3 Object by Truffle.

You can search about it. But if you think it'll be good and you know the tool, let's do it!! We can make an issue to re-write the tests usin Web3 1.0
